### ISSUE TITLE
Support connection strings for Azure-backed modelrepo

### ISSUE THREAD

**Issue by Massendefekt** [triton-inference-server/server#4072]
**Is your feature request related to a problem? Please describe.**

Currently, connections to model repositories backed by azure storage can only be authenticated via environment variables. This is suboptimal in cases where triton is embedded in another application or model repositories are changed dynamically (both is the case in my application).

**Describe the solution you'd like**
Add support for [Azure Storage connection strings](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string). 

In the case of the C API, a call to `TRITONSERVER_ServerOptionsSetModelRepositoryPath` would then accept a such a connection string. Configuration via environment variables could be kept, but might alternatively also accept a connection string.

**Describe alternatives you've considered**
Overriding environment variables via `setenv` from `libc` works, but is definitely not a clean solution.

**Additional context**
I'm happy to contribute this feature in case you support its addition. :)

**Comment by dyastremsky** [comment]
Thank you for submitting this feature request and offering to contribute it. We've created an internal ticket to track it and will get back to you soon.

**Comment by deadeyegoodwin** [comment]
@CoderHam  I think this is a feature we may be generally interested in. Can we support a contribution?

**Comment by CoderHam** [comment]
Yes we can support a contribution. @Massendefekt would you be able to add this enhancement with some guidance?
